frontends/parsers/p4/p4parser.ypp

```diff
@@ -152,9 +152,11 @@ inline std::ostream& operator<<(std::ostream& out, const Token& t) {
 
 using namespace P4;
 
-#define YYLLOC_DEFAULT(Cur, Rhs, N)                                             \
-    ((Cur) = (N) ? YYRHSLOC(Rhs, 1) + YYRHSLOC(Rhs, N)                          \
-                 : Util::SourceInfo(driver.sources, YYRHSLOC(Rhs, 0).getEnd()))
+#define YYLLOC_DEFAULT(Cur, Rhs, N)                                               \
```

Comment:
@ChrisDodd I tried to fix some crashes that happen because the error reporter is accessing an uninitialized source info. It can also happen that the parsers produce a 0:0 source info, which is invalid. Not quite clear to me why. Might be that that source info is temporary and it does not matter.

Any idea why? Is that expected?

Reply:
The `0:0` source info is an 'invalid' source info, generated by the default ctor and corresponds to there being no source location for something.  All cases that access that should deal with it -- the error reporter special-cases it to not print anything source related.  It should not crash.

Reply:
The error report will crash if the sources object is nullptr, which can happen when you create an invalid source info. I fixed some cases up in the program structure translator.